src/exports-named.ts

```diff
@@ -0,0 +1,55 @@
+import Hls from './hls';
+import { Events } from './events';
+import { ErrorTypes, ErrorDetails } from './errors';
+import { Level } from './types/level';
+import AbrController from './controller/abr-controller';
+import AudioTrackController from './controller/audio-track-controller';
+import AudioStreamController from './controller/audio-stream-controller';
+import BasePlaylistController from './controller/base-playlist-controller';
+import BaseStreamController from './controller/base-stream-controller';
+import BufferController from './controller/buffer-controller';
+import CapLevelController from './controller/cap-level-controller';
+import CMCDController from './controller/cmcd-controller';
+import ContentSteeringController from './controller/content-steering-controller';
+import EMEController from './controller/eme-controller';
+import ErrorController from './controller/error-controller';
+import FPSController from './controller/fps-controller';
+import SubtitleTrackController from './controller/subtitle-track-controller';
+
+export {
+  Hls,
```

Comment:
@robwalch With this, ``npm run build:ci`` passes locally. I also replaced the ``dist`` folder into the ``node_modules/hls.js/dist`` folder of my project [jellyfin-vue](https://github.com/jellyfin/jellyfin-vue).

By changing this line: https://github.com/jellyfin/jellyfin-vue/blob/master/frontend/src/components/Playback/PlayerElement.vue#L33 to:

``
import Hls, { ErrorData, Events, ErrorTypes } from 'hls.js';
``

and fixing all the linter errors reported in that file, my project build without any issues,.

So doing this change is indeed the only required thing to be fully backwards-compatible.

```suggestion
export default Hls;

export {
  Hls,
```

Reply:
By the way, you can quickly test this in my project with GH Codespaces, since we have first-class support for devcontainers.

"Code" button at the right corner > Create Codespace

Just cd into the ``frontend`` directory and run ``npm run build`` to build. Go to the previously mentioned file where hls.js is imported and do the relevant tests.

Reply:
Fantastic! The mixed named and default exports was only a problem with the UMD output which we've given it's own target so it remains unchanged. So happy that the ESM output does not make the default export an object * with a default property.